packages/regenesis-surgery/scripts/data.ts

```diff
@@ -95,6 +95,39 @@ export const loadSurgeryData = async (
     configs = loadConfigs()
   }
 
+  // Get a reference to an ETH (mainnet) provider.
+  console.log('Connecting to ETH provider...')
+  const ethProvider = new ethers.providers.JsonRpcProvider(
+    configs.ethProviderUrl
+  )
+  // Create a mainnet wallet to have access to the chainid call.
+  // This is not used elsewhere so just use the ropstenPrivateKey
+  const mainnetWallet = new ethers.Wallet(
+    configs.ropstenPrivateKey,
+    ethProvider
+  )
+  const mainnetChainId = await mainnetWallet.getChainId()
+  if (mainnetChainId !== 1) {
+    throw new Error(
+      `Mainnet chain id incorrect, got ${mainnetChainId} and expected 1`
+    )
+  }
+
+  // Get a reference to the L2 provider so we can load pool data.
+  // Do validation on the chain id before reading data from disk
+  // because that is slow
+  console.log('Connecting to L2 provider...')
+  const l2Provider = new ethers.providers.JsonRpcProvider(configs.l2ProviderUrl)
+  const l2Wallet = new ethers.Wallet(configs.ropstenPrivateKey, l2Provider)
+  const l2ChainId = await l2Wallet.getChainId()
+  if (configs.l2NetworkName === 'mainnet' && l2ChainId !== 10) {
```

Comment:
Let's just automatically pick the network name based on the chain ID

Reply:
Thats a better idea